projects/plugins/jetpack/modules/masterbar/admin-menu/class-base-admin-menu.php

```diff
@@ -161,12 +164,41 @@ public function update_submenus( $slug, $submenus_to_update ) {
 			return;
 		}
 
-		foreach ( $submenu[ $slug ] as $i => $submenu_item ) {
-			if ( array_key_exists( $submenu_item[2], $submenus_to_update ) ) {
-				$submenu_item[2] = $submenus_to_update[ $submenu_item[2] ];
-				// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
-				$submenu[ $slug ][ $i ] = $submenu_item;
+		/**
+		 * Iterate over all submenu items and add the hide the submenus with CSS classes.
+		 * This is done separately of the second foreach because the position of the submenu might change.
+		 */
+		foreach ( $submenu[ $slug ] as $index => $item ) {
+			if ( ! array_key_exists( $item[2], $submenus_to_update ) ) {
+				continue;
+			}
+
+			$css_classes = empty( $item[4] ) ? 'hide-if-js hide-if-no-js' : $item[4] . ' hide-if-js hide-if-no-js';
+
+			// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+			$submenu [ $slug ][ $index ][4] = $css_classes;
+		}
```

Comment:
Can we reuse `hide_submenu_page` here?

Reply:
No, because we are doing that check to see if the slug is part of the $submenus_to_update. We could refactor a bit I think by having another method like this:
```
public function hide_submenu_element( $index, $slug, $item ) {
			$css_classes = empty( $item[4] ) ? self::HIDE_CSS_CLASS : $item[4] . ' ' . self::HIDE_CSS_CLASS;

			// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
			$submenu [ $slug ][ $index ][4] = $css_classes;
}
```

This way we'll have the hide logic in one place. Nice catch, thanks :)